### PR TITLE
#2 Add yearbook sales box

### PR DESCRIPTION
**Pull request by sylgas**

### CODE REVIEW

File: src/actions/salesEvents.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export function fetchSales() {
+    return {
+        type: FETCH_SALES
+    }
+}
+
+export function fetchSalesSuccess(sales) {
+    return {
+        type: FETCH_SALES_SUCCESS,
+        payload: {sales: sales}
```
Comment (whitcik) [inline src/actions/salesEvents.js]: You could use shorter ES6 syntax
`payload: { sales }`

File: src/actions/salesEvents.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    }
+}
+
+export function fetchSalesError(error) {
+    return {
+        type: FETCH_SALES_SUCCESS,
+        payload: {error: error}
```
Comment (whitcik) [inline src/actions/salesEvents.js]: ditto as above

File: src/actions/salesEvents.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {FETCH_SALES, FETCH_SALES_SUCCESS} from "../constants/actions";
+
+export function fetchSales() {
```
Comment (whitcik) [inline src/actions/salesEvents.js]: I recommend es6 syntax, for example:
`export const fetchSales = () => ({`
    ` type: FETCH_SALES`
`});`

File: src/components/Common/Bubble.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import sizeMe from 'react-sizeme';
+import PropTypes from 'prop-types';
+
+const Bubble = ({leftBubbleMargin, text, size}) => {
+    const halfWidth = size.width / 2;
+    const leftMargin = leftBubbleMargin - halfWidth;
+    const dashOutLeftMargin = halfWidth - 5;
+    const dashInLeftMargin = dashOutLeftMargin + 1;
+    return (
+        <div className="bubble" style={{marginLeft: leftMargin}}>
```
Comment (whitcik) [inline src/components/Common/Bubble.js]: I think you can rename const leftMargin to marginLeft and use shorter es6 syntax `{marginLeft}`

File: src/components/YearbookSales/SalesBox/index.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    return [
+        pad(date.getMonth() +1, 1),
+        pad(date.getDate(), 2),
+        pad(date.getFullYear(), 4),
+    ].join('/');
+}
+
+function pad(numberString, size) {
+    let padded = numberString.toString();
```
Comment (whitcik) [inline src/components/YearbookSales/SalesBox/index.js]: `padded `should be const

File: src/components/YearbookSales/SalesBoxChart/SalesBoxChart.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import "./salesBoxChart.less"
+import * as React from "react";
+import sizeMe from 'react-sizeme';
+import Bubble from "../../Common/Bubble";
+
+class SalesBoxChart extends React.Component {
```
Comment (whitcik) [inline src/components/YearbookSales/SalesBoxChart/SalesBoxChart.js]: This component could be stateless

File: src/components/YearbookSales/SalesBoxChart/SalesBoxChart.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    render() {
+        const {max, campus, online, size} = this.props;
+        const width = size.width - 26;
```
Comment (whitcik) [inline src/components/YearbookSales/SalesBoxChart/SalesBoxChart.js]: what is 26? I think this should be constants, declared above component or if you will be use this constants in another file, constans should be declared in external file. 
for example:
`const MARGIN_WIDTH = 26`

File: src/components/YearbookSales/SalesBoxChart/SalesBoxChart.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        const part = width / max;
+        const campusWidth = max ? campus * part : 0;
+        const onlineWidth = max ? online * part : 0;
+        const leftBubbleMargin = campusWidth + onlineWidth + 13;
```
Comment (whitcik) [inline src/components/YearbookSales/SalesBoxChart/SalesBoxChart.js]: As above for 13

File: src/api/salesApi.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+            max: 1500
+        }
+    }
+};
+
+export function fetchSales(){
+    console.log('[API] fetchSales');
+    return new Promise((resolve, reject) => {
+        setTimeout(() => {
+            resolve(Object.assign({}, response.data));
```
Comment (whitcik) [inline src/api/salesApi.js]: You don't need return new reference here, becouse you do it in salesReducer.js. btw. data fetched from API always will be have new reference.

File: src/actions/salesEvents.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+export function fetchSalesSuccess(sales) {
+    return {
+        type: FETCH_SALES_SUCCESS,
+        payload: {sales: sales}
+    }
+}
+
+export function fetchSalesError(error) {
+    return {
+        type: FETCH_SALES_SUCCESS,
```
Comment (whitcik) [inline src/actions/salesEvents.js]: i think type should be `FETCH_SALES_ERROR`

File: .babelrc [hunk 11]
```diff
@@ -0,0 +1,10 @@
+{
+  "env": {
+    "test": {
+      "presets": [
+        "es2015",
+        "react"
+      ]
+    }
+  }
+}
```
Comment (bigsondev) [inline .babelrc]: You have it already in webpack, it's probably not necessary to have additional .babelrc file. Webpack rule should be enough.

File: src/components/YearbookSales/SalesBox/SalesBox.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        return (
+            <div className="box">
+                <div className="box-header clearfix">
+                    <h2>Yearbook Sales</h2>
+                    <span className="pull-right"><b>Final Sales: </b><a>{this.props.finalSales}</a></span>
+                </div>
+                <div className="box-header clearfix">
+                    <span className="pull-left"><b>Personalization: </b>{this.props.personalization}</span>
+                    <span className="pull-right"><b>Exact Quantity: </b>{this.props.quantity}</span>
+                </div>
```
Comment (bigsondev) [inline src/components/YearbookSales/SalesBox/SalesBox.js]: It should be one box-header class: Header/Content/Footer.

File: src/components/YearbookSales/SalesBox/index.js (hunk 5 above)
Comment (bigsondev) [reply src/components/YearbookSales/SalesBox/index.js]: formatDate and pad should be somewhere in helper methods.

**Pull request merged**